executor/executor_required_rows_test.go

```diff
@@ -900,37 +892,37 @@ func (s *testExecSuite) TestVecGroupCheckerDATARACE(c *C) {
 		case mysql.TypeJSON:
 			chk.Column(0).ReserveJSON(1)
 			j := new(json.BinaryJSON)
-			c.Assert(j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 123, 123))), IsNil)
+			require.NoError(t, j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 123, 123))))
 			chk.Column(0).AppendJSON(*j)
 		}
 
 		_, err := vgc.splitIntoGroups(chk)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 
 		switch mType {
 		case mysql.TypeVarString:
-			c.Assert(vgc.firstRowDatums[0].GetString(), Equals, "abc")
-			c.Assert(vgc.lastRowDatums[0].GetString(), Equals, "abc")
+			require.Equal(t, "abc", vgc.firstRowDatums[0].GetString())
+			require.Equal(t, "abc", vgc.lastRowDatums[0].GetString())
 			chk.Column(0).ReserveString(1)
 			chk.Column(0).AppendString("edf")
-			c.Assert(vgc.firstRowDatums[0].GetString(), Equals, "abc")
-			c.Assert(vgc.lastRowDatums[0].GetString(), Equals, "abc")
+			require.Equal(t, "abc", vgc.firstRowDatums[0].GetString())
+			require.Equal(t, "abc", vgc.lastRowDatums[0].GetString())
 		case mysql.TypeNewDecimal:
-			c.Assert(vgc.firstRowDatums[0].GetMysqlDecimal().String(), Equals, "123")
-			c.Assert(vgc.lastRowDatums[0].GetMysqlDecimal().String(), Equals, "123")
+			require.Equal(t, "123", vgc.firstRowDatums[0].GetMysqlDecimal().String())
+			require.Equal(t, "123", vgc.lastRowDatums[0].GetMysqlDecimal().String())
 			chk.Column(0).ResizeDecimal(1, false)
 			chk.Column(0).Decimals()[0] = *types.NewDecFromInt(456)
-			c.Assert(vgc.firstRowDatums[0].GetMysqlDecimal().String(), Equals, "123")
-			c.Assert(vgc.lastRowDatums[0].GetMysqlDecimal().String(), Equals, "123")
+			require.Equal(t, "123", vgc.firstRowDatums[0].GetMysqlDecimal().String())
+			require.Equal(t, "123", vgc.lastRowDatums[0].GetMysqlDecimal().String())
 		case mysql.TypeJSON:
-			c.Assert(vgc.firstRowDatums[0].GetMysqlJSON().String(), Equals, `{"123": 123}`)
-			c.Assert(vgc.lastRowDatums[0].GetMysqlJSON().String(), Equals, `{"123": 123}`)
+			require.Equal(t, `{"123": 123}`, vgc.firstRowDatums[0].GetMysqlJSON().String())
+			require.Equal(t, `{"123": 123}`, vgc.lastRowDatums[0].GetMysqlJSON().String())
 			chk.Column(0).ReserveJSON(1)
 			j := new(json.BinaryJSON)
-			c.Assert(j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 456, 456))), IsNil)
+			require.Nil(t, j.UnmarshalJSON([]byte(fmt.Sprintf(`{"%v":%v}`, 456, 456))))
```

Comment:
This line should be `NoError`, too.

Reply:
I've submit a patch to fix all of this for @feitian124 :)

Reply:
thanks